wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -634,10 +659,526 @@ public boolean testWUQuery() throws Exception
+     * @return WUQueryResponse
+     */
+    private WUQueryResponse wuqueryresponseConvert(org.hpccsystems.ws.client.gen.wsworkunits.v1_56.WUQueryResponse in)
+    {
+    	WUQueryResponse out = new WUQueryResponse();
+    	
+
+    	if(in.getType() != null) out.setType(in.getType());
+    	if(in.getCluster() != null) out.setCluster(in.getCluster());
+        if(in.getRoxieCluster() != null) out.setRoxieCluster(in.getRoxieCluster());
+        if(in.getOwner() != null) out.setOwner(in.getOwner());
+        if(in.getState() != null) out.setState(in.getState());
+        if(in.getStartDate() != null) out.setStartDate(in.getStartDate());
+        if(in.getEndDate() != null) out.setEndDate(in.getEndDate());
+        if(in.getECL() != null) out.setECL(in.getECL());
+        if(in.getJobname() != null) out.setJobname(in.getJobname());
+        if(in.getLogicalFile() != null) out.setLogicalFile(in.getLogicalFile());
+        if(in.getLogicalFileSearchType() != null) out.setLogicalFileSearchType(in.getLogicalFileSearchType());
+        if(in.getCurrent() != null) out.setCurrent(in.getCurrent());
+        if(in.getNext() != null) out.setNext(in.getNext());
+        if(in.getCount() != null) out.setCount(in.getCount());
+        if(in.getPageSize() != null) out.setPageSize(in.getPageSize());
+        if(in.getPrevPage() != null) out.setPrevPage(in.getPrevPage());
+        if(in.getNextPage() != null) out.setNextPage(in.getNextPage());
+        if(in.getLastPage() != null) out.setLastPage(in.getLastPage());
+        if(in.getNumWUs() != null) out.setNumWUs(in.getNumWUs());
+        if(in.getFirst() != null) out.setFirst(in.getFirst());
+        if(in.getPageStartFrom() != null) out.setPageStartFrom(in.getPageStartFrom());
+        if(in.getPageEndAt() != null) out.setPageEndAt(in.getPageEndAt());
+        if(in.getLastNDays() != null) out.setLastNDays(in.getLastNDays());
+        if(in.getSortby() != null) out.setSortby(in.getSortby());
+        if(in.getDescending() != null) out.setDescending(in.getDescending());
+        if(in.getBasicQuery() != null) out.setBasicQuery(in.getBasicQuery());
+        if(in.getFilters() != null) out.setFilters(in.getFilters());
+        if(in.getCacheHint() != null) out.setCacheHint(in.getCacheHint());
+        if(in.getExceptions() != null) out.setExceptions(arrayofespexceptionsConvert(in.getExceptions()));
+    	if(in.getWorkunits() != null)
+    	{
+    		ECLWorkunit[] tmpwu = new ECLWorkunit[in.getWorkunits().length];
+    		for(int i = 0; i < in.getWorkunits().length; i++)
+    			tmpwu[i] = eclworkunitConvert(in.getWorkunits()[i]);
+    		out.setWorkunits(tmpwu);	
+    	}
+    	
+    	return out;
+    }
+    
+    /**
+     * Check the version found at instatiation with compatibility version
+     * 
+     * @param Version input
+     * @return boolean
+     * 	true if server build version >= input version
+     */
+    private boolean compatibilityCheck(Version input)
+    {
+    	if(version.major > input.major)
```

Comment:
no.  If target version major > input major, return as compatible, if less than, return incompatible, if equal, try the next level of precision.